Added Sun_checks Checkstyle with custom supression script and other integrations \ Danshal

levilime opened the pull request:
User Story: As a programmer I want to make informed decisions on the quality of a merge request by using Continuous integration.

Because the connector is mostly code not written or maintained by us, we'll only do checkstyle on the code that we touch. In this way we provide strong arguments that we can do checkstyle well (by example) while not running in constantly correct checkstyle mistakes from the repo we fork off. This would be an uphill battle.

The python script on the server is run by using jython. This is because Travis only supports one development language per integration.

Cobertura fails when the total line coverage for the Environment is lower than 75%.

It would be advisible to check the code with checkstyle before commiting. You can run the eclipse plugin with the official Sun style and then only correct the style for the lines you've edited.

PMD and Findbugs are just outputted in the Travis Log.

levilime committed:
Added rudimentary checkstyle for connector

levilime committed:
Use Jython to run python stylesuppress code

levilime committed:
Added comments to travis and diff script

levilime committed:
Added comment for magic number diff script

And deleted cool.java . That shouldn't be there.

levilime committed:
git diff in Travis now takes all changes between HEAD and master

levilime committed:
Fixed bug in diff script and only account for .java and started work on cobertura

Commit:
Added PMD

Commit:
Test

Commit:
?

Commit:
How bout now?

Commit:
removed some stuf

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Fixed?

Commit:
Update .travis.yml

Commit:
Update .travis.yml

levilime committed:
Added cobertura that fails the test when coverage for env is too low

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

levilime committed:
Cobertura only runs in travis through cobertura:check

levilime committed:
PMD outputs results in CI

levilime committed:
Fixed merge conflicts with travis of context branch

levilime committed:
Added findbugs report to travis

levilime committed:
Integrated checkstyle and Cobertura, PMD and Findbugs is printed in CI

levilime committed:
Make Travis not fail if coverage is low cobertura.

But it shows in the build log. This is due to other classes in the environment not being sufficiently tested yet by the repo forked upon.

levilime committed:
Cobertura halt on failure

with total linerate and branchrate.

levilime committed:
Only check total line coverage cobertura

levilime committed:
newest context branch changes in this feature branch

levilime committed:
Merge branch 'CobPMDCheck' of github.com:levilime/tygron into CobPMDBugsChecknew

levilime committed:
Diff compares this commit with context branch eishub/tygron

NickTUD committed:
Merge pull request #11 from levilime/CobPMDBugsChecknew

Added Google_checks Checkstyle with custom supression script and other integrations

spiderbiggen commented:
This looks good. will be very useful to have the CI only run these checks on our changes. and not on everything

JoshCode commented:
I suggest that the lines 

```
notifications:
  slack: $slack_token
```

from #4 are added so that when pulling this .travis.yml into our own repositories, Slack integration will still work and this .travis.yml will never have to be edited by other groups anymore, which could result in overwriting things, because different groups that want Travis CI integration will have different tokens, so when pushing commits, the new .travis.yml will get pushed too, resulting in invalid tokens being pulled to other team's repo's.

\- CodeFox [![CodeFox](https://avatars1.githubusercontent.com/u/18596181?v=3&s=32)](https://github.com/codefoxnl)

spiderbiggen commented:
@JoshCode what would happen here if you don't add a slack token to your variables on travis? nothing?

MaxGroenenboom commented:
Should all groups use the same checkstyle rules?

JoshCode commented:
@spiderbiggen Yeah, nothing should happen. Perhaps a warning, but not a failure.

levilime commented:
@MaxGroenenboom I think it is logical all groups should use the same checkstyle, because we'll all use the same integration (that on eishub/tygron context) This pull request is also about which checkstyle we should use together. I think google checkstyle is a good one.

levilime commented:
@JoshCode isn't it possible to integrate Travis with a git repo through the webhooks, and it being connected to your group fork? Is it necessary to have this token there? But I can add it if it is useful to you or others :smile:

levilime committed:
Update .travis.yml with slack token.

JoshCode commented:
@levilime It is for the Slack - Travis CI integration, which is only possible with a token, or we'd have to write an entire integration bot ourselves ;)

MaxGroenenboom commented:
As I said during the meeting, we (WhySoSerious) don't agree yet on the choice of the checkstyle format, just leaving it here as well for the record.

spiderbiggen commented:
@MaxGroenenboom but then what do you as a group suggest

MaxGroenenboom commented:
What I would suggest as a person (so not speaking for my group right now):

A format that looks a bit more like what the connector is right now. The Google format yields over 2500 warnings, while the Sun Checks (Eclipse) configuration yields only over 500 warnings. This means if we use the Sun Checks (Eclipse) configuration, our added code is much more consistent with the existing code, which is more important than our code looking "nice" according to some random company.

levilime commented:
@MaxGroenenboom I like your thinking, I think you yield good arguments to change it to Sun checkstyle. I will change it to

MaxGroenenboom commented:
Also, is there any good reason to use spaces instead of tabs? I know a lot of software engineers prefer using spaces but I never heard a good reason for it, and as the entire connector currently uses tabs instead of spaces, I think it would be better to use tabs for the same reason I wrote in my last comment on this issue.

By the way, Sun Checks has a modified configuration for Eclipse (which is made conform the default Eclipse styling configuration), which is the one I recommended. This configuration by default uses tabs so we could use it? I think most people use Eclipse anyway so I see no reason not to.

levilime committed:
Replaced google_checks by sun_checks for checkstyle config

levilime commented:
@MaxGroenenboom OLD The sun checkstyle configuration is now used. It doesn't fail on tabs that I know of OLD. Oops, nope that's the FileTabCharacter property. I think you have nice convincing arguments for no tabs so will take that from the sun checkstyle.

levilime committed:
Update sun_checks.xml to not check on tabs.

Because of the argument that these are already used continuously in the existing code base and tabs are not hurting anyone right now.

JoshCode commented on README.md:
```diff
@@ -1,5 +1,7 @@
 # Tygron EIS connector
 
-Tygron is an EIS environment that connects with the tygron http://www.tygron.com system for urban planning.
+Tygron is an EIS environment that connects with the tygron http://www.tygron.com sytem fsor urban planning.
```
Typo

levilime committed:
Changed checkstyle on travis to the check goal.

Multiple fixes have been made to now actually let the build fail when
there is a checkstyle error. Checkstyle is invoked a different way and
it is invoked with arguments for the location of the custom_style file
and the suppression file, because it didn't find it before and used the
default.

Also the checkstylediff is adjusted to make it work better due to
presenting the file locations differently.

Lastly the supression markup is fixed to reflect the specifications of
the xml sheet used. Also the custom style file is adusted somewhat because
there were specification errors there before.

levilime committed:
tygronchange

levilime committed:
Made diff window for checkstyling smaller.

Now really only the lines that were edited compared
to the current state of the eishub/tygron context branch
will be style checked in the CI.

levilime committed:
Ommitted suppression file from custom style.

And added print feedback in checkstylediff file to receive
information on what files in what lines are checkstyled.

And fixed typo in checkstylediff.

levilime committed:
Fixed print feedback in checkstylediff

And some more cleanup in the code.

levilime commented:
Please merge :point_right: :red_circle:

spiderbiggen commented:
This looks good please merge \ Rampestampers

Denpeer commented:
Looks like it's ready to merge \ WhySoSerious
